### PR TITLE
Event is active field, Delete the event

### PR DESCRIPTION
**Pull request by kmpol**
Before the (partial - no tests) implementation:

Delete event button: no behavior

After the (partial - no tests) implementation:

Delete button deletes in two way:
-- First: If there are event attenders - soft delete (remove from view, but still available in the db)
-- Second: If there are no event attenders - hard delete (remove from db)

Before the deletion - modal asks if you are sure about the deletion.

### CODE REVIEW

File: core/src/main/java/greencity/webcontroller/ManagementEventsController.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class ManagementEventsController {
      */
     @GetMapping
     public String getAllEvents(@RequestParam(required = false, name = "query") String query, Model model,
-        @ApiIgnore Pageable pageable, EventViewDto eventViewDto) {
+        @ApiIgnore Pageable pageable, EventViewDto eventViewDto, @RequestParam(required = false, name = "active", defaultValue = "false") boolean active) {
```
Comment (marcinkunert) [inline core/src/main/java/greencity/webcontroller/ManagementEventsController.java]: If you use defaultValue then `required = false` is not needed

File: core/src/main/resources/static/css/sidebar.css [hunk 2]
```diff
@@ -199,7 +199,18 @@ block:hover img{
     color: white;
 }
 
+button.close {
+    padding: 2rem 1rem !IMPORTANT;
```
Comment (marcinkunert) [inline core/src/main/resources/static/css/sidebar.css:203]: I've never seen `!important` written in caps 😅

File: core/src/main/resources/templates/core/management_events.html [hunk 3]
```diff
@@ -355,29 +384,29 @@ <h4 class="modal-title">[[#{greenCity.events.page.edit.events}]]</h4>
...
+<!--                            <input type="button" class="btn btn-default btn-tertiary" data-dismiss="modal"-->
+<!--                                   th:value="#{greenCity.pages.popup.cancel.}">-->
+<!--                            <a href="" id="deleteOneSubmit" class="btn btn-remove">-->
+<!--                                <div>[[#{greenCity.pages.delete}]]</div>-->
+<!--                            </a>-->
+<!--                        </div>-->
+<!--                    </form>-->
+<!--                </div>-->
+<!--            </div>-->
+<!--        </div>-->
```
Comment (marcinkunert) [inline core/src/main/resources/templates/core/management_events.html]: Did you comment that out on purpose? Either delete it completely or bring this back

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 4]
```diff
@@ -76,6 +77,7 @@ public EventDto save(AddEventDtoRequest addEventDtoRequest, String email,
     }
 
     @Override
+    @Transactional
```
Comment (marcinkunert) [inline service/src/main/java/greencity/service/EventServiceImpl.java]: If this is added on purpose, what's the benefit?

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 5]
```diff
@@ -136,6 +138,7 @@ private void setSubscribes(Page<Event> events, PageableAdvancedDto<EventDto> eve
 
     private PageableAdvancedDto<EventDto> buildPageableAdvancedDto(Page<Event> eventsPage) {
         List<EventDto> eventDtos = eventsPage.stream()
+                .filter(Event::getIsActive)
```
Comment (marcinkunert) [inline service/src/main/java/greencity/service/EventServiceImpl.java]: It is better to do filtering on the database side.

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 6]
```diff
@@ -254,6 +257,21 @@ public EventVO findById(Long eventId) {
...
+        //TODO: remove duplicated code
+        Set<EventAttenderDto> allEventAttenders = getAllEventAttenders(eventId);
+        UpdateEventDto eventUpdates = new UpdateEventDto();
+        eventUpdates.setId(eventId);
+        eventUpdates.setIsActive(false);
+        if(allEventAttenders.isEmpty()){
+            delete(eventId, (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal());
+        } else {
+            update(eventUpdates, (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal(), null);
+        }
```
Comment (marcinkunert) [inline service/src/main/java/greencity/service/EventServiceImpl.java]: I'm not a fan of this approach. Consider moving the principal part into delete and update metods

**Review by marcinkunert — CHANGES_REQUESTED** [review]

File: service/src/main/java/greencity/service/EventServiceImpl.java (hunk 5 above)
Comment (kmpol) [reply service/src/main/java/greencity/service/EventServiceImpl.java]: Yes, but there might be in the future some functionality, which could be "filter", that shows me only active and inactive events.
For now we are getting whole db and filter inactive events out. In the future we still want whole db downloaded and just filter active out in order to show inactive ones. So we could active them back.

**Review by PawelMiskowiec — APPROVED** [review]

**Pull request closed without merging**